### PR TITLE
feat: upgrade @storybook/react 6.5.16 with a11y addon (accessbility

### PR DESCRIPTION
**Pull request by vipinWednesday**
### Ticket Link

---
### Description
* Upgrade  to @storybook/react 6.5.16
* Modified  `.storybook` Folder  
* Added a11y addon and test-runner.
* Update stories for Header , Clickable and T components 
---

### Steps to Reproduce / Test
---
*  Storybook :`yarn storybook`
*  Test the accessbility : `yarn test-storybook`

---

### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's
(Updated)
typogr2

---

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,12 +88,15 @@
   },
   "pre-commit": "lint:staged",
   "dependencies": {
+    "@babel/preset-typescript": "^7.18.6",
```
Comment (praveenkumar1798) [inline package.json:91]: Why do we need this ?

Comment (vipinWednesday) [reply package.json:91]: @storybook/test-runner need this .

File: package.json [hunk 2]
```diff
@@ -141,14 +148,20 @@
...
-    "@storybook/addon-actions": "^5.2.1",
-    "@storybook/addon-knobs": "^5.2.1",
-    "@storybook/addon-links": "^5.2.1",
-    "@storybook/addons": "^5.2.1",
-    "@storybook/react": "^5.2.1",
+    "@storybook/addon-a11y": "^6.5.16",
+    "@storybook/addon-actions": "^6.5.16",
+    "@storybook/addon-essentials": "^6.5.16",
+    "@storybook/addon-interactions": "^6.5.16",
+    "@storybook/addon-links": "^6.5.16",
```
Comment (praveenkumar1798) [inline package.json:155]: Why are there additional addons ?
@storybook/addon-actions, @storybook/addon-knobs, @storybook/addon-links iwere the ones that were there earlier. Lets just keep them unless we are doing something with the new plugins that are added

File: package.json [hunk 3]
```diff
@@ -87,19 +89,24 @@
...
+    "@storybook/addons": "^6.5.16",
     "apisauce": "2.1.1",
     "chalk": "4.1.1",
     "compression": "1.7.4",
+    "concurrently": "^7.6.0",
     "cross-env": "7.0.3",
     "dotenv": "10.0.0",
     "express": "4.16.4",
     "history": "4.9.x",
+    "http-server": "^14.1.1",
```
Comment (praveenkumar1798) [inline package.json:109]: all the new ones that you have added here are dev dependencies. Can we move it to dev dependencies section ?

File: package.json (hunk 1 above)
Comment (praveenkumar1798) [reply package.json:92]: okay then should be moved to dev dependency

**Comment by praveenkumar1798** [comment]
@vipinWednesday Please add this to the documentation as well. And I guess there needs to be some changes to the storybook files we are reffering to in the doc because those are changed now

File: package.json (hunk 2 above)
Comment (vipinWednesday) [reply package.json:155]: @storybook/knobs is deprecated  rest are added by storybook itself.

File: package.json [hunk 4]
```diff
@@ -93,6 +95,8 @@
     "@mui/material": "^5.11.5",
     "@mui/styled-engine": "^5.11.0",
     "@mui/styled-engine-sc": "^5.11.0",
+    "@storybook/addon-controls": "^6.5.16",
+    "@storybook/addons": "^6.5.16",
```
Comment (praveenkumar1798) [inline package.json:99]: Why do we need this here ?

Comment (praveenkumar1798) [reply package.json:99]: Move to dev dependencies

Comment (vipinWednesday) [reply package.json:99]: Storybook did that by default i think , i'll try to move it to dev and test.

**Review by praveenkumar1798 — APPROVED** [review]

**Pull request merged**